src/sql/base/browser/ui/listBox/listBox.ts

```diff
@@ -140,11 +149,23 @@ export class ListBox extends SelectBox {
 
 		for (let i = 0; i < indexes.length; i++) {
 			this.selectElement.remove(indexes[i]);
+			this.options.splice(indexes[i], 1);
 		}
+		super.setOptions(this.options);
 	}
 
 	public add(option: string): void {
-		this.selectElement.add(this.createOption(option));
+		let optionObj = this.createOption(option);
+		this.selectElement.add(optionObj);
+
+		// make sure that base options are updated since that is used in selection not selectElement
+		this.options.push(optionObj);
+		super.setOptions(this.options);
+	}
+
+	public setOptions(options: ISelectOptionItem[], selected?: number): void {
+		this.options = options;
```

Comment:
It would be great to have tests for this kind of stuff (adding/removing items from a list box) - especially for stuff like this that extends VS controls since those may change at any point. I haven't looked at how feasible that is without resorting to full on UI testing but hopefully we should be able to mock up enough stuff to access the items stored by the underlying SelectBox.

Reply:
yeah, will be good to have tests for these controls all up, but out of scope for this PR.